src/test/java/core/basesyntax/service/impl/TransactionProcessorServiceImplTest.java

```diff
@@ -0,0 +1,112 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransactionProcessorService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class TransactionProcessorServiceImplTest {
+    private Map<String, Integer> fruitCounts;
+    private List<FruitTransaction> transactions = new ArrayList<>();
+    private TransactionProcessorService transactionProcessorService;
+    private TransactionProcessorServiceImpl singleTransaction;
+
+    private Map<FruitTransaction.Operation, OperationStrategy> strategies;
+
+    @BeforeEach
+    void setUp() {
+        fruitCounts = new HashMap<>();
+        strategies = new HashMap<>();
+        singleTransaction = new TransactionProcessorServiceImpl(strategies);
+        transactionProcessorService = new TransactionProcessorServiceImpl(strategies);
+        FruitTransaction transaction1 =
+                new FruitTransaction(
+                        FruitTransaction.Operation.PURCHASE,
+                        "apple",
+                        10);
+        FruitTransaction transaction2 = new FruitTransaction(
+                FruitTransaction.Operation.BALANCE,
+                "banana",
+                5);
+        transactions = List.of(transaction1, transaction2);
+    }
+
+    @Test
+    void processTransaction_validInput_success() {
+        Map<String, Integer> expected =
+                transactionProcessorService.processTransaction(transactions);
+        Map<String, Integer> actual = Storage.getFruits();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void processTransaction_invalidOperationType() {
+        FruitTransaction.Operation operation = FruitTransaction.Operation.RETURN;
+        String fruit = "apple";
+        int quantity = 10;
+        FruitTransaction transaction = new FruitTransaction(operation, fruit, quantity);
+
+        singleTransaction.processSingleTransaction(transaction);
+        Assertions.assertNull(Storage.getFruits().get(fruit));
```

Comment:
it's not ok that nothing happens, it will make it hard to debug, if some of operations got ignored from very long list, it's better to have the program fail and fix them one by one, instead of searching for them on your own

Reply:
I decided to delete this test because in the implementation, the processTransaction method in src/main uses the processSingleTransactions method, which handles invalid operation types. Therefore, it seems redundant to me